server/memphis_handlers_monitoring.go

```diff
@@ -928,46 +928,100 @@ cleanup:
 	return throughputs, nil
 }
 
-func (mh MonitoringHandler) GetMainOverviewData(c *gin.Context) {
-	stationsHandler := StationsHandler{S: mh.S}
-	stations, totalMessages, totalDlsMsgs, err := stationsHandler.GetAllStationsDetails(false)
-	if err != nil {
-		serv.Errorf("GetMainOverviewData: GetAllStationsDetails: " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-		return
-	}
-	systemComponents, metricsEnabled, err := mh.GetSystemComponents()
-	if err != nil {
-		if strings.Contains(strings.ToLower(err.Error()), "cannot connect to the docker daemon") {
-			serv.Warnf("GetMainOverviewData: GetSystemComponents: " + err.Error())
-			c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "Failed getting system components data: " + err.Error()})
-		} else {
-			serv.Errorf("GetMainOverviewData: GetSystemComponents: " + err.Error())
-			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+func (mh MonitoringHandler) getMainOverviewDataAsync(c *gin.Context) (models.MainOverviewData, error) {
```

Comment:
getMainOverviewDataAsync -> getMainOverviewDataDetails and it should not get the c as a parameter